nnpdf_data/nnpdf_data/theory.py

```diff
@@ -0,0 +1,52 @@
+"""
+This module provides an unique source and definition for all the possible parameters
+that a theory card can contain. 
+It also implement some utilities function to manage the theory cards. 
+"""
+
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class TheoryCard:
+    ID: int
+    PTO: int
+    FNS: str
+    DAMP: int
+    DAMPPOWERc: None #Power of the damping factor in FONLL for the c
+    DAMPPOWERb: None #Power of the damping factor in FONLL for the b
+    IC: int
+    ModEv: str
+    ModSV: None
+    XIR: float
+    XIF: float
+    NfFF: int
+    nfref: 5
+    nf0: 3
+    Q0: float
+    alphas: float
+    Qref: float
+    QED: int
+    alphaqed: float
+    HQ: str
+    mc: float
+    Qmc: float
+    kcThr: float
+    mb: float
+    Qmb: float
+    kbThr: float
+    mt: float
+    Qmt: float
+    ktThr: float
+    CKM: list[float]
+    MZ: float
+    MW: float
+    GF: float
+    SIN2TW: float
+    TMC: int
+    MP: float
+    Comments: str
+    FactScaleVar: None #used in yadism to allow for Fact scale var
+    RenScaleVar: None #used in yadism to allow for Ren scale var
```

Comment:
```suggestion
    FactScaleVar: bool = False #used in yadism to allow for Fact scale var
    RenScaleVar: bool = False #used in yadism to allow for Ren scale var
```

But if it is used only for yadism I don't think it should be part of the theory card, rather the program calling yadism (`pinefarm`) should be able to guess these from `xif/xir`? No?
(or... why is it not always set to True, we want to save that information when we create grids...)

Reply:
In my opinion, this information should not be guessed from `xir` and `xif` because I agree with you that we should always ask for scale variations (i.e. they should be True by default). However, this is a default we should enforce in `yadism` or `pinefarm` and, once we do it there, we can remove these two parameters from the theory card

Reply:
So this is used to save time. `RenScaleVar: true` can be used to generate all the scale variations even if `xir: 1.0`, 
so the 2 keys are decoupled. 
If you want to set it automatically, this should be true if the theory is associated to scale varied theories [here](https://github.com/NNPDF/nnpdf/blob/master/validphys2/src/validphys/scalevariations/scalevariationtheoryids.yaml)

Reply:
Well, as in the comment above, the theory parameters here are mixed with whatever program configurations (I would have limited theory to `MZ`, `alphas_ref`, and similar).

In any case, yadism is not special in any way, at least from the point of view of Pineko (and whatever else is downstream). Since you're working with PineAPPL grids, you only care about the structure.

However, in general I agree about the decoupling, and the option could be useful to include or not the scale variations in the grid, for whatever MC.
Not all the MC will have the option in their API, but it could also be implemented (and, if they are always including scale variations, they could be dropped from the grid in Pinefarm, after generation).